tools/ExampleExtractor/ExtractorAndTesterUsersGuide.md

```diff
@@ -0,0 +1,483 @@
+
+````
+> <!-- Example: {template:"standalone-lib", name:"UsingAliasDirectives11", ignoredWarnings:["CS0169"]} -->
+````
+
+Here’s an *example_annotation* showing both expected and ignored warnings:
+
+````
+> <!-- Example: {template:"standalone-lib", name:"TryCatchFinally", ignoredWarnings:["CS0219"], expectedWarnings:["CS0162"]} -->
+````
+
+### Expected Runtime Output
+
+During execution, some test applications are expected to write one or more lines of output to the console. In such cases, the list of expected output-line strings is provided by the *expected_output* annotation directive, so the two sets of text can be compared, verbatim, for equality. This *annotation_directive* is optional. (This directive makes no provision for output written to a file.)
+
+```ANTLR
+expected_output
+    : 'expectedOutput' ':' '[' output_string (',' output_string)* ']'
+    ;
+output_string
+    : JSON_string_value
+    ;
+```
+
+Consider the following example, which writes two lines to the console:
+
+````
+> <!-- Example: {template:"standalone-console", name:"ScopeGeneral3", expectedOutput:["hello, world", "A"]} -->
+> ```csharp
+> using System;
+>
+> class A {}
+>
+> class Test
+> {
+>     static void Main()
+>     {
+>         string A = "hello, world";
+>         string s = A;                      // expression context
+>         Type t = typeof(A);                // type context
+>         Console.WriteLine(s);              // writes "hello, world"
+>         Console.WriteLine(t);              // writes "A"
+>     }
+> }
+> ```
+````
+
+### Expected Exception
+
+An expected exception can be identified and checked for. This *annotation_directive* is optional.
+
+```ANTLR
+expected_exception
+    : 'expectedException' ':' exception_name
+    ;
+exception_name
+    : JSON_string_value    // an unqualified typename
+    ;
+```
+
+Consider the following example:
+
+````
+> <!-- Example: {template:"standalone-console", name:"CovarianceException", expectedException:"ArrayTypeMismatchException"} -->
+> ```csharp
+> class Test
+> {
+>     static void Fill(object[] array, int index, int count, object value) 
+>     {
+>         for (int i = index; i < index + count; i++)
+>         {
+>             array[i] = value;
+>         }
+>     }
+>
+>     static void Main() 
+>     {
+>         string[] strings = new string[100];
+>         Fill(strings, 0, 100, "Undefined");
+>         Fill(strings, 0, 10, null);
+>         Fill(strings, 90, 10, 0);  // throws System.ArrayTypeMismatchException
+>     }
+> }
+> ```
+````
+
+## Other Information
+
+### Unsafe Code
+
+All templates support compilation of examples containing unsafe code.
+
+### Examples Containing Pseudo-Code
+
+Some examples contain C# grammar-rule names, which show a general format rather than source code that will compile. Such names are fenced using the following notation: `«rule_name»`. These examples do *not* have *example_annotation*s.
```

Comment:
These typically aren't "examples" though - they're parts of the standard explaining how it processes things. That's why I'm not too bothered by it not working with the example parser :)

Reply:
Agreed.